pkg/validatorfactory/factory_test.go

```diff
@@ -0,0 +1,65 @@
+package validatorfactory
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/kubectl-validate/pkg/openapiclient"
+	"sigs.k8s.io/yaml"
+)
+
+func TestValidatorFactory_TestPatcher(t *testing.T) {
+	tests := []struct {
+		name  string
+		file  string
+		typed interface{}
+		want  interface{}
+	}{{
+		name:  "configmap",
```

Comment:
This tests the change against a native type `ConfigMap`, but I thought the motivating example for this change was with CRDs so I expected to see a test for a CRD.

Reply:
Oh OK, i did just that in https://github.com/kyverno/kyverno/pull/8384, i will make the same test here.

Reply:
Added a unit test with a crd